### PR TITLE
FIXED JENKINS-6290 Add "New View" link on "My Views" page

### PR DESCRIPTION
**Pull request by Austry**
See [JENKINS-6290](https://issues.jenkins-ci.org/browse/JENKINS-6290).

### Desired reviewers

@jenkinsci/hacktoberfest

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
So it is another part of https://github.com/jenkinsci/jenkins/pull/3086

File: core/src/main/java/hudson/model/MyViewsProperty.java [hunk 1]
```diff
@@ -224,7 +224,7 @@ public ViewsTabBar getViewsTabBar() {
 
     public List<Action> getViewActions() {
         // Jenkins.getInstance().getViewActions() are tempting but they are in a wrong scope
-        return Collections.emptyList();
+        return Collections.singletonList(new NewViewAction());
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/MyViewsProperty.java]: Does it work in such way? I see no routing logic to `ModifiableViewGroup#addView`

Comment (Austry) [reply core/src/main/java/hudson/model/MyViewsProperty.java]: yes, it's work
https://yadi.sk/i/JZQWXkP33Nvd7N
probably becouse of line in ```View#getActions``` 
```  	result.addAll(getOwner().getViewActions()); ```

**Comment by daniel-beck** [comment]
@oleg-nenashev do you need to make a team public, or why is the hacktoberfest mention not working?

**Comment by daniel-beck** [comment]
What is the relation of this change to #3084?

**Comment by oleg-nenashev** [comment]
@daniel-beck The team is public. No idea why it does not org for non-orgmembers.

> What is the relation of this change to #3084?

For starters they refer to the same issue, but add "New View" links in different places (global views and user views). The latter PR is https://github.com/jenkinsci/jenkins/pull/3086 now

**Comment by Austry** [comment]
Oh, sorry
Issue was not assigned to anyone in JIRA, so I thought I can take it

**Comment by oleg-nenashev** [comment]
@Austry I see no collision between changes you propose. It's fine to have two PRs in parallel, though you could definitely sync-up with @tinexw

**Comment by Austry** [comment]
Ok 
In @tinexw solution new link is added in global views. I don't think that link is so important to be on every screen, so I added it only to "My Views" page.
I think it's a question of user experience
So if you think what it should be on every page, then we need to merge #3086 and close this one

**Comment by daniel-beck** [comment]
@oleg-nenashev I edited the original post without content changes to make the team mention work.

Haven't found a reference in GitHub help, but apparently mentions only work for members of the team's organization. I've seen a similar problem with `jenkinsci/code-reviewers` in the past.

**Review by oleg-nenashev — APPROVED** [review]
I am fine with adding this button in the UserProperty, but IMHO we need to make it consistent. It's not the first time the existing button confuses people.

**Comment by oleg-nenashev** [comment]
@jenkinsci/code-reviewers any opinions?

File: core/src/main/java/hudson/model/MyViewsProperty.java [hunk 2]
```diff
@@ -254,7 +254,23 @@ public String getIconFileName() {
...
 			return "/me/my-views";
 		}
-		
+
+    }
+
+    private static class NewViewAction implements Action {
+        @Override
+        public String getIconFileName() {
+            return "new-document.png";
```
Comment (jglick) [inline core/src/main/java/hudson/model/MyViewsProperty.java]: Clashes with use of `folder` in https://github.com/jenkinsci/jenkins/pull/3086.

Comment (Austry) [reply core/src/main/java/hudson/model/MyViewsProperty.java]: ok, I'll change it to folder
but ```new-document``` seems more natural for me

Comment (jglick) [reply core/src/main/java/hudson/model/MyViewsProperty.java]: I do not have any particular opinion, just noting that two PRs seem to be proposing conflicting UIs.

**Comment by oleg-nenashev** [comment]
@Austry we have integrated #3086 (see the list of the reasons in comments there). How would you prefer to proceed with this PR? We could probably just close it (won't affect your hacktoberfest stats) or reintegrate it with master and add more features on the top of #3086 if possible

**Comment by Austry** [comment]
#3086 covers all features from this pr
so I close it
thx anyway @oleg-nenashev

**Pull request closed without merging**